### PR TITLE
Add ReplicationController metrics

### PR DESCRIPTION
**Pull request by dominikschulz**
This PR adds ReplicationController metrics.

---
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kube-state-metrics/56)

### CODE REVIEW

File: replicationcontroller.go [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+)
+
+var (
+	descReplicationControllerStatusReplicas = prometheus.NewDesc(
+		"kube_replicationcontroller_status_replicas",
+		"The number of replicas per ReplicationController.",
+		[]string{"namespace", "replicationcontroller"}, nil,
+	)
+	descReplicationControllerStatusFullyLabeledReplicas = prometheus.NewDesc(
+		"kube_replicationcontroller_status_fully_labeled_replicas",
```
Comment (brancz) [inline replicationcontroller.go]: docs missing for this one

File: replicationcontroller.go [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+		"The number of fully labeled replicas per ReplicationController.",
+		[]string{"namespace", "replicationcontroller"}, nil,
+	)
+	descReplicationControllerStatusReadyReplicas = prometheus.NewDesc(
+		"kube_replicationcontroller_status_ready_replicas",
```
Comment (brancz) [inline replicationcontroller.go]: docs missing for this one

File: README.md [hunk 3]
```diff
@@ -72,6 +72,19 @@ additional metrics!
 | kube_pod_container_limits_cpu_cores | Gauge | `container`=&lt;container-name&gt; <br> `pod`=&lt;pod-name&gt; <br> `namespace`=&lt;pod-namespace&gt; <br> `node`=&lt; node-address&gt; |
 | kube_pod_container_limits_memory_bytes | Gauge | `container`=&lt;container-name&gt; <br> `pod`=&lt;pod-name&gt; <br> `namespace`=&lt;pod-namespace&gt; <br> `node`=&lt; node-address&gt; |
 
+### ReplicationController metrics
+
+| Metric name| Metric type | Labels/tags |
+| ---------- | ----------- | ----------- |
+| kube_replicationcontroller_status_replicas | Gauge | `replicationcontroller`=&lt;replicationcontroller-name&gt; <br> `namespace`=&lt;replicationcontroller-namespace&gt; |
+| kube_replicationcontroller_status_replicas_available | Gauge | `replicationcontroller`=&lt;replicationcontroller-name&gt; <br> `namespace`=&lt;replicationcontroller-namespace&gt; |
```
Comment (brancz) [inline README.md]: `.Status.AvailableReplicas` does not exist for the ReplicationController object.

File: README.md [hunk 4]
```diff
@@ -72,6 +72,19 @@ additional metrics!
...
+| kube_replicationcontroller_status_replicas_unavailable | Gauge | `replicationcontroller`=&lt;replicationcontroller-name&gt; <br> `namespace`=&lt;replicationcontroller-namespace&gt; |
+| kube_replicationcontroller_status_replicas_updated | Gauge | `replicationcontroller`=&lt;replicationcontroller-name&gt; <br> `namespace`=&lt;replicationcontroller-namespace&gt; |
+| kube_replicationcontroller_status_replicas_observed_generation | Gauge | `replicationcontroller`=&lt;replicationcontroller-name&gt; <br> `namespace`=&lt;replicationcontroller-namespace&gt; |
```
Comment (brancz) [inline README.md]: implementation missing for this one

File: README.md [hunk 5]
```diff
@@ -72,6 +72,19 @@ additional metrics!
...
 | kube_pod_container_limits_memory_bytes | Gauge | `container`=&lt;container-name&gt; <br> `pod`=&lt;pod-name&gt; <br> `namespace`=&lt;pod-namespace&gt; <br> `node`=&lt; node-address&gt; |
 
+### ReplicationController metrics
+
+| Metric name| Metric type | Labels/tags |
+| ---------- | ----------- | ----------- |
+| kube_replicationcontroller_status_replicas | Gauge | `replicationcontroller`=&lt;replicationcontroller-name&gt; <br> `namespace`=&lt;replicationcontroller-namespace&gt; |
+| kube_replicationcontroller_status_replicas_available | Gauge | `replicationcontroller`=&lt;replicationcontroller-name&gt; <br> `namespace`=&lt;replicationcontroller-namespace&gt; |
+| kube_replicationcontroller_status_replicas_unavailable | Gauge | `replicationcontroller`=&lt;replicationcontroller-name&gt; <br> `namespace`=&lt;replicationcontroller-namespace&gt; |
+| kube_replicationcontroller_status_replicas_updated | Gauge | `replicationcontroller`=&lt;replicationcontroller-name&gt; <br> `namespace`=&lt;replicationcontroller-namespace&gt; |
```
Comment (brancz) [inline README.md]: `.Status.UpdatedReplicas` does not exist for the `ReplicationController` object.

File: README.md [hunk 6]
```diff
@@ -72,6 +72,19 @@ additional metrics!
 | kube_pod_container_limits_cpu_cores | Gauge | `container`=&lt;container-name&gt; <br> `pod`=&lt;pod-name&gt; <br> `namespace`=&lt;pod-namespace&gt; <br> `node`=&lt; node-address&gt; |
 | kube_pod_container_limits_memory_bytes | Gauge | `container`=&lt;container-name&gt; <br> `pod`=&lt;pod-name&gt; <br> `namespace`=&lt;pod-namespace&gt; <br> `node`=&lt; node-address&gt; |
 
+### ReplicationController metrics
+
+| Metric name| Metric type | Labels/tags |
+| ---------- | ----------- | ----------- |
+| kube_replicationcontroller_status_replicas | Gauge | `replicationcontroller`=&lt;replicationcontroller-name&gt; <br> `namespace`=&lt;replicationcontroller-namespace&gt; |
+| kube_replicationcontroller_status_replicas_available | Gauge | `replicationcontroller`=&lt;replicationcontroller-name&gt; <br> `namespace`=&lt;replicationcontroller-namespace&gt; |
+| kube_replicationcontroller_status_replicas_unavailable | Gauge | `replicationcontroller`=&lt;replicationcontroller-name&gt; <br> `namespace`=&lt;replicationcontroller-namespace&gt; |
```
Comment (brancz) [inline README.md]: `.Status.UnavailableReplicas` does not exist for the ReplicationController object.

File: README.md [hunk 7]
```diff
@@ -72,6 +72,19 @@ additional metrics!
...
+| kube_replicationcontroller_status_replicas_updated | Gauge | `replicationcontroller`=&lt;replicationcontroller-name&gt; <br> `namespace`=&lt;replicationcontroller-namespace&gt; |
+| kube_replicationcontroller_status_replicas_observed_generation | Gauge | `replicationcontroller`=&lt;replicationcontroller-name&gt; <br> `namespace`=&lt;replicationcontroller-namespace&gt; |
+| kube_replicationcontroller_spec_replicas | Gauge | `replicationcontroller`=&lt;replicationcontroller-name&gt; <br> `namespace`=&lt;replicationcontroller-namespace&gt; |
+| kube_replicationcontroller_spec_paused | Gauge | `replicationcontroller`=&lt;replicationcontroller-name&gt; <br> `namespace`=&lt;replicationcontroller-namespace&gt; |
```
Comment (brancz) [inline README.md]: `Paused` does not exist for the `ReplicationController` object.

**Comment by tmegow** [comment]
Where is this at? ReplicationController metrics would be great, how can I help?

**Comment by brancz** [comment]
@tmegow this pull request is quite outdated. If you are looking to contribute this, it's probably best if you start over and pick bits and pieces out of this one as there were some larger restructurings of this project. Generally speaking a PR for `ReplicationController` metrics would be very welcome.

**Comment by dominikschulz** [comment]
Sorry for the delay. Rebased and updated my branch.

**Comment by brancz** [comment]
Great thanks for fixing! lgtm :+1:

**Pull request merged**